tgui/packages/tgui/interfaces/PreferencesMenu/MainPage.tsx

```diff
@@ -511,11 +513,22 @@ export const MainPage = (
           delete nonContextualPreferences['name_is_always_random'];
         }
 
-        return (
-          <>
-            {multiNameInputOpen && (
+        // a little bit hardcoded section to display preference categories in tabs and bottom section
+        // prefInfoTab switch should take keys from INFOTAB_LIST(in DM define)
+        let infotab_content;
+        switch (prefInfoTab) {
+          case 'non_contextual': // sadly, it's not possible to bring constant define from DM. We should handle manually.
+            infotab_content = (
```

Comment:
switch statements with assignments like this absolutely wreck rehydration (re-rendering of the UI) sometimes.

Instead, add all of the tabs to the overall UI and then change their style from `display: none` to `display: block` depending on if they are selected.